app/controllers/users_controller.rb

```diff
@@ -22,9 +25,50 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by(id: params[:id])
+  end
+
+  def update
+    @user = User.find_by(id: params[:id])
+    if @user.update(user_params)
+      flash[:success] = t "ud_prf"
+      redirect_to @user
+    else
+      render "edit"
+    end
+  end
+
+  def index
+    @users = User.paginate(page: params[:page])
+  end
+
+  def destroy
+    user = User.find_by(id: params[:id])
+    if user&.destroy
+      flash[:success] = t "User_deleted"
+    else
+      flash[:danger] = t "Delete_fail"
+    end
+    redirect_to users_url
+  end
+
   private
   def user_params
     params.require(:user).permit :name, :email, :password,
                                  :password_confirmation
   end
+
+  def logged_in_user
+    return if logged_in?
+
+    store_location
+    flash[:danger] = t "plesase_lg"
+    redirect_to login_url
+  end
+
+  def correct_user
+    @user = User.find_by(id: params[:id])
+    redirect_to(root_url) unless current_user?(@user)
```

Comment:
thêm flash cho người ta biết lý do nha
```suggestion
    ...
    return if current_user?(@user)
    
    flash ... # báo lỗi không có quyền truy cập
    redirect_to(root_url) 
```